cobc/codegen.c

```diff
@@ -7532,230 +7553,256 @@ output_call (struct cb_call *p)
 		if (name_is_literal_or_prototype) {
 			s = get_program_id_str (p->name);
 			name_str = cb_encode_program_id (s, 1, cb_fold_call);
-			lookup_call (name_str);
-			callname = s;
+			/* Distinguishing lookup from call*/
+			if(
+#ifdef HAVE_JNI 
+				strncmp("Java.", s, 6) == 0
+#else
+				0
+#endif
+			) {
+				lookup_java_call(s + 6);
+				output_line ("if (call_java_%s == NULL || cob_glob_ptr->cob_physical_cancel)", name_str);
+				output_block_open();
+				output_prefix();
+				output ("call_java_%s = ", name_str);
+				char *first_dot = strchr(s + 6, '.');
+				if (first_dot != NULL) {
+					*first_dot = '\0';
+					char *method_name = first_dot + 1;
+					char *class_name = s;
+					output("cob_resolve_java (\"%s\", \"%s\", \"()V\");", class_name, method_name);
+				}
+				output ("cob_resolve_java ();");
+				output_newline ();
+				output_block_close ();
```

Comment:
Right here you need to generate a call to `cob_call_java(…)` with the appropriate handle name `call_java_%s`. Arguments will come in a subsequent PR.